code/modules/atmospherics/components/omni_devices/filter.dm

```diff
@@ -188,44 +180,54 @@
 			if("switch_mode")
 				switch_mode(dir_flag(href_list["dir"]), mode_return_switch(href_list["mode"]))
 			if("switch_filter")
-				var/new_filter = input(usr,"Select filter mode:","Change filter",href_list["mode"]) in list("None", "Oxygen", "Nitrogen", "Carbon Dioxide", "Nitrous Oxide", "Hydrogen")
-				switch_filter(dir_flag(href_list["dir"]), mode_return_switch(new_filter))
+				var/list/gas_list = get_gas_names()
+				var/new_filter = input(usr,"Select filter mode:","Change filter",href_list["mode"]) in gas_list
+				switch_filter(dir_flag(href_list["dir"]), ATM_FILTER, get_decl_from_symbol(new_filter))
 
 	update_icon()
 	SSnano.update_uis(src)
 	return
 
+/obj/machinery/atmospherics/omni/filter/proc/get_gas_names()
+	var/list/output = list()
+	var/list/gases = decls_repository.get_decls_of_subtype(/decl/material/gas)
+	for(var/G in gases)
+		if(G == /decl/material/gas/alien)
+			continue
+		var/decl/material/gas/GE = gases[G]
+		output += GE.gas_symbol
+	return output
+
+/obj/machinery/atmospherics/omni/filter/proc/get_decl_from_symbol(var/sym)
+	var/list/gases = decls_repository.get_decls_of_subtype(/decl/material/gas)
```

Comment:
I think it would be better to have materials cache themselves by symbol during init rather than limiting it to iterating `/decl/material/gas` - ideally in the future we won't use types for categorization.